libusb/libusb.h

```diff
@@ -953,6 +957,96 @@ struct libusb_ss_usb_device_capability_descriptor {
 	uint16_t bU2DevExitLat;
 };
 
+/** \ingroup libusb_desc
+ *  enum used in \ref libusb_ssplus_sublink_attribute
+ */
+enum libusb_superspeedplus_sublink_attribute_sublink_type {
+	LIBUSB_SSPLUS_ATTR_TYPE_SYM = 0,
+	LIBUSB_SSPLUS_ATTR_TYPE_ASYM = 1,
+};
+
+/** \ingroup libusb_desc
+ *  enum used in \ref libusb_ssplus_sublink_attribute
+ */
+enum libusb_superspeedplus_sublink_attribute_sublink_direction {
+	LIBUSB_SSPLUS_ATTR_DIR_RX = 0,
+	LIBUSB_SSPLUS_ATTR_DIR_TX = 1,
+};
+	
+/** \ingroup libusb_desc
+ *  enum used in \ref libusb_ssplus_sublink_attribute
+ */
+enum libusb_superspeedplus_sublink_attribute_exponent {
+	LIBUSB_SSPLUS_ATTR_EXP_BIT = 0,
+	LIBUSB_SSPLUS_ATTR_EXP_KIB = 1,
+	LIBUSB_SSPLUS_ATTR_EXP_MIB = 2,
+	LIBUSB_SSPLUS_ATTR_EXP_GIG = 3,
```

Comment:
This is what we get from the specs:

```
This field defines the
base 10 exponent times 3, that shall be applied to the
Lane Speed Mantissa (LSM) when calculating the
maximum bit rate represented by this Lane Speed
Attribute.

0 Bits per second
1 Kb/s
2 Mb/s
3 Gb/s
``

So it seems to be base 10, and bits. I have remove the I to be less confusing.

Reply:
The added comment is quite helpful, but the names are still ambiguous because uppercase 'B' still can be confused for byte.  Why not just be fully descriptive:

```diff
-LIBUSB_SSPLUS_ATTR_EXP_BIT = 0,
-LIBUSB_SSPLUS_ATTR_EXP_KB = 1,
-LIBUSB_SSPLUS_ATTR_EXP_MB = 2,
-LIBUSB_SSPLUS_ATTR_EXP_GG = 3,
+LIBUSB_SSPLUS_ATTR_EXP_BITSPERSEC = 0,
+LIBUSB_SSPLUS_ATTR_EXP_KILOBITSPERSEC = 1,
+LIBUSB_SSPLUS_ATTR_EXP_MEGABITSPERSEC = 2,
+LIBUSB_SSPLUS_ATTR_EXP_GIGABITSPERSEC = 3,
```

Reply:
These names suggested a too long. I think the current names along with the comment in the header are enough for user to figure out the units.

Reply:
Too long for what? Code is read more than code is written, better to optimize for reading. Also, autocomplete makes writing faster than it used to be.

The comment in the header is only in the header, it does not help the reader when seeing these constants used in some source file.

Reply:
How much do you want to put in the name? Some may argue that the name you suggest `LIBUSB_SSPLUS_ATTR_EXP_KILOBITSPERSEC` is still ambiguous since it does not specify if kilo = 1000 or kilo = 1024.

So should it be `LIBUSB_SSPLUS_ATTR_EXP_KILO_BASE10_BITSPERSEC`?

Pretty much everybody know that USB speed are `Xbps`, And if they have a doubt they can read the enum comment in `libusb.h`.

Reply:
>How much details should we put in the name?

Enough to understand what the thing is. `LIBUSB_SSPLUS_ATTR_EXP_KB` does not tell me if K=kilo or K=kibi or B=bit or B=byte. It also doesn't indicate that it's 'per second' vs an amount of memory.

>...is still ambiguous since it does not specify if kilo = 1000 or kilo = 1024.

Some may argue that, but the debate was settled: kilo = 1000, kibi = 1024.

> Every USB speed figure are advertised in `Xbps` which I would expect USB dev to be famiilar with.

I honestly didn't know. I know little about USB in fact, libusb is my abstraction.

Reply:
and `LIBUSB_SSPLUS_ATTR_EXP_GG` is especially weird. To me, GG means 'good game'.

Reply:
"GG" is obviously a typo, since it says "GB" in the comment above the code.

Anyway these values refer to electric signaling rates on the wires, related to clock rates and other frequencies which are always SI units and powers of 10. Commonly you will see bps (bit per second) and kbs (kilobit per second) used for these units. So capitalized they would be BPS, KBS, MBS, GBS. (They are not about data storage size (or higher level data transfer rates) where "bytes" (and storage sizes in MiB or KiB) could possibly be used.)

This would also be consistent with the text in the specification, just omitting the slash and capitalized: _Bits per second, Kb/s, Mb/s, Gb/s._

I don't see any reason to deviate from these conventions to save one character, neither to pack more of the specification into the enum names.

Reply:
Changed the enum according to your comment

```
enum libusb_superspeedplus_sublink_attribute_exponent {
	LIBUSB_SSPLUS_ATTR_EXP_BITS = 0,
	LIBUSB_SSPLUS_ATTR_EXP_KBS = 1,
	LIBUSB_SSPLUS_ATTR_EXP_MBS = 2,
	LIBUSB_SSPLUS_ATTR_EXP_GBS = 3,
};
```

Reply:
BITS?

Reply:
It has to include`per second`? So `BIT/S` becomes `BITS`.
All other contant end in S to specify the rate.

What about you tell me exactly what you want for these constant and I will happily use that?

Reply:
I was thinking BPS, like I wrote above.

Reply:
Done. The enum now says:

```
enum libusb_superspeedplus_sublink_attribute_exponent {
	LIBUSB_SSPLUS_ATTR_EXP_BPS = 0,
	LIBUSB_SSPLUS_ATTR_EXP_KBS = 1,
	LIBUSB_SSPLUS_ATTR_EXP_MBS = 2,
	LIBUSB_SSPLUS_ATTR_EXP_GBS = 3,
};
```

Is there anything else blocking this PR from being merged?

Reply:
Thanks. I don't think there is anything else needed from your side. I still would like to hear from @hjelmn on #1429 though.